packages/agreement-process/test/computeAgreementsState.test.ts

```diff
@@ -0,0 +1,606 @@
+      suspendedByPlatform: true,
+      suspendedByConsumer: false,
+      suspendedByProducer: false,
+    };
+
+    const nonUpdatableAgreement = {
+      ...getMockAgreement(
+        eservice1.id,
+        consumer.id,
+        randomArrayItem([agreementState.archived, agreementState.rejected])
+      ),
+      descriptorId: eservice1.descriptors[0].id,
+      producerId: eservice1.producerId,
+    };
+
+    await addOneEService(eservice1);
+    await addOneEService(eservice2);
+    await addOneAgreement(updatableAgreement1);
+    await addOneAgreement(updatableAgreement2);
+    await addOneAgreement(nonUpdatableAgreement);
+
+    await agreementService.computeAgreementsStateByAttribute(
+      randomArrayItem([
+        tenantCertifiedAttribute.id,
+        tenantDeclaredAttribute.id,
+        tenantVerifiedAttribute.id,
+      ]),
+      consumer,
+      {
+        authData,
+        serviceName: "",
+        correlationId: "",
+        logger: genericLogger,
+      }
+    );
+
+    const nonUpdatableAgreementStateUpdateEvent = await readLastAgreementEvent(
+      nonUpdatableAgreement.id
+    );
+
+    expect(nonUpdatableAgreementStateUpdateEvent).toMatchObject({
+      type: "AgreementAdded",
+      event_version: 2,
+      version: "0",
+      stream_id: nonUpdatableAgreement.id,
+    });
+
+    const agreement1StateUpdateEvent = await readLastAgreementEvent(
+      updatableAgreement1.id
+    );
+
+    expect(agreement1StateUpdateEvent).toMatchObject({
+      type: "AgreementPutInDraftByPlatform",
+      event_version: 2,
+      version: "1",
+      stream_id: updatableAgreement1.id,
+    });
+
+    const agreement1StateUpdateEventData = decodeProtobufPayload({
+      messageType: AgreementPutInDraftByPlatformV2,
+      payload: agreement1StateUpdateEvent.data,
+    });
+
+    expect(agreement1StateUpdateEventData).toMatchObject({
+      agreement: toAgreementV2({
+        ...updatableAgreement1,
+        state: agreementState.draft,
+        suspendedByPlatform: false,
+      }),
+    });
+
+    const agreement2StateUpdateEvent = await readLastAgreementEvent(
+      updatableAgreement2.id
+    );
+
+    expect(agreement2StateUpdateEvent).toMatchObject({
+      type: "AgreementUnsuspendedByPlatform",
+      event_version: 2,
+      version: "1",
+      stream_id: updatableAgreement2.id,
+    });
+
+    const agreement2StateUpdateEventData = decodeProtobufPayload({
+      messageType: AgreementUnsuspendedByPlatformV2,
+      payload: agreement2StateUpdateEvent.data,
+    });
+
+    expect(agreement2StateUpdateEventData).toMatchObject({
+      agreement: toAgreementV2({
+        ...updatableAgreement2,
+        state: agreementState.active,
+        suspendedByPlatform: false,
+      }),
+    });
+  });
+
+  it("suspends an Agreement by platform even when the Agreement is already suspended but with suspendedByPlatform = false", async () => {
+    // Tests the following state transitions:
+    // - Suspended -> Suspended
```

Comment:
this should be clear by the test description (if it's not feel free to update the description)